src/Compilers/CSharp/Portable/Symbols/Metadata/PE/TupleTypeDecoder.cs

```diff
@@ -85,10 +85,19 @@ private TupleTypeDecoder(ImmutableArray<string> elementNames)
             // bad metadata
             if (hasTupleElementNamesAttribute && elementNames.IsDefaultOrEmpty)
             {
-                return new UnsupportedMetadataTypeSymbol();
+                return TypeSymbolWithAnnotations.Create(new UnsupportedMetadataTypeSymbol(), isNullableIfReferenceType: null);
             }
 
-            return DecodeTupleTypesInternal(metadataType, elementNames, hasTupleElementNamesAttribute);
+            var decoded = DecodeTupleTypesInternal(metadataType.TypeSymbol, elementNames, hasTupleElementNamesAttribute);
+            return TypeSymbolWithAnnotations.Create(decoded, isNullableIfReferenceType: metadataType.IsNullable, metadataType.CustomModifiers);
+        }
+
+        public static TypeSymbol DecodeTupleTypesIfApplicable(
```

Comment:
>DecodeTupleTypesIfApplicable [](start = 33, length = 28)

What's the point of this method?

Reply:
Oops, I forgot to remove this method. But in the meantime, I've decided to repurpose it.

---
In reply to: [194197256](https://github.com/dotnet/roslyn/pull/27589#discussion_r194197256) [](ancestors = 194197256)